services/QuillLMS/app/services/classlink_integration/instructor_linker.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+module ClasslinkIntegration
+  class InstructorLinker < ApplicationService
+    attr_reader :classlink_instance_id, :persisted_user, :school_sourced_id
+
+    def initialize(persisted_user:, classlink_instance_id:, school_sourced_id:)
+      @classlink_instance_id = classlink_instance_id
+      @persisted_user = persisted_user
+      @school_sourced_id = school_sourced_id
+    end
+
+    def run
+      return unless persisted_user&.teacher?
+      return unless school_sourced_id
+      return unless classlink_instance_school
+
+      persisted_user.role == persisted_user.admin? ? create_admin_linkage : create_teacher_linkage
```

Comment:
Is this comparison the right one?  Is `admin?` not a boolean that we're comparing against the string value of `role`?